src/state/common.rs

```diff
@@ -27,13 +27,23 @@ impl Order {
             total_buy: Fr::zero(),
         }
     }
+}
+
+impl Order {
     pub fn hash(&self) -> Fr {
         let mut data = Fr::zero();
         data.add_assign(&self.status);
         data.add_assign(&shl(&self.tokenbuy, 32));
         data.add_assign(&shl(&self.tokensell, 64));
         hash(&[data, self.filled_sell, self.filled_buy, self.total_sell, self.total_buy])
     }
+    pub fn is_filled(&self) -> bool {
```

Comment:
This logic isn't quite right.

Reply:
https://github.com/Loopring/protocols/blob/5e49e7b10e80238fbf71822e2e1bf110af686872/packages/loopring_v3/circuit/Gadgets/MatchingGadgets.h#L224-L257